src/library/scala/util/hashing/MurmurHash3.scala

```diff
@@ -101,32 +101,85 @@ private[hashing] class MurmurHash3 {
     h = mixLast(h, c)
     finalizeHash(h, n)
   }
-  /** Compute a hash that depends on the order of its arguments.
-   */
+
+  /** Compute a hash that depends on the order of its arguments. Potential range
+    * hashes are recognized to produce a hash that is compatible with rangeHash.
+    */
   final def orderedHash(xs: IterableOnce[Any], seed: Int): Int = {
     var n = 0
     var h = seed
+    var rangeState = 0 // 0 = no data, 1 = first elem read, 2 = has valid diff, 3 = invalid
+    var rangeDiff = 0
+    var prev = 0
+    var initial = 0
     xs.iterator foreach { x =>
```

Comment:
Why not use the iterator bare here (i.e. with `hasNext` and `next`)?  The logic should be identical to the while loop at that point.

Reply:
Yes, and it is *much* faster, too.